scuttlebutt/src/failure_detector.rs

```diff
@@ -69,20 +72,27 @@ impl FailureDetector {
             debug!(node_id = node_id, phi = phi, "updating node liveliness");
             if phi > self.config.phi_threshold {
                 self.live_nodes.remove(node_id);
+                self.dead_nodes.insert(node_id.to_string());
                 // Remove current sampling window so that when the node
                 // comes back online, we start with a fresh sampling window.
                 self.node_samples.write().unwrap().remove(node_id);
             } else {
                 self.live_nodes.insert(node_id.to_string());
+                self.dead_nodes.remove(node_id);
             }
         }
     }
 
-    /// Returns a list of living nodes.
+    /// Returns a list of live nodes.
     pub fn live_nodes(&self) -> impl Iterator<Item = &str> {
```

Comment:
We almost always end up collecting into a set. Should we return a `HashSet<NodeId>` directly for `live_nodes`, `dead_nodes`, `seed_nodes`?